web/src/inbox_ui.js

```diff
@@ -518,10 +532,11 @@ export function collapse_or_expand(container_id) {
     }
     $toggle_icon.toggleClass("icon-collapsed-state");
 
-    if (collapsed_containers.has(container_id)) {
-        collapsed_containers.delete(container_id);
+    const ls_set_to_modify = is_muted ? expanded_muted_streams : collapsed_containers;
+    if (ls_set_to_modify.has(container_id)) {
+        ls_set_to_modify.delete(container_id);
     } else {
-        collapsed_containers.add(container_id);
+        ls_set_to_modify.add(container_id);
     }
```

Comment:
I'm skeptical this code is correct; certainly it's hard to check. I think we want to arrange to have a `is_collapsed` variable, and just handle each of the cases for different values of that and `is_muted` explicitly; it'll be a bit more code but way easier to reason about.

I also suspect the current logic may leak some state in the case that the muting status for a stream is changed; so probably it's possible for a stream to be in both `expanded_muted_streams` and `collapsed_containers`. I don't know if there's a real problem that can cause.